xcube/core/subsampling.py

```diff
@@ -69,13 +71,16 @@ def subsample_dataset(
                 assert slices is not None
                 new_var = var[slices]
             else:
-                dim = dict()
-                if x_name in var.dims:
-                    dim[x_name] = step
-                if y_name in var.dims:
-                    dim[y_name] = step
-                var_coarsen = var.coarsen(dim=dim, boundary="pad", coord_func="min")
-                new_var: xr.DataArray = getattr(var_coarsen, agg_method)()
+                if agg_method == "mode":
+                    new_var: xr.DataArray = _mode(var, x_name, y_name, step)
+                else:
+                    dim = dict()
+                    if x_name in var.dims:
+                        dim[x_name] = step
+                    if y_name in var.dims:
+                        dim[y_name] = step
+                    var_coarsen = var.coarsen(dim=dim, boundary="pad", coord_func="min")
+                    new_var: xr.DataArray = getattr(var_coarsen, agg_method)()
```

Comment:
Extract function `_agg_builtin` and declare return type `xr.DataArray`. In accordance to helper `_agg_mode` below.